src/main/java/com/google/devtools/build/remote/client/ActionGrouping.java

```diff
@@ -153,6 +153,29 @@ void printByAction(PrintWriter out) throws IOException {
     }
   }
 
+  void printByActionJson() throws IOException {
+    System.out.println("[");
```

Comment:
There should be a streaming json iteration mechanism that can print this with a stock json printer. Constructing output like this leaves things very brittle and in a situation where no guarantee of proper json output for outliers is considered.

https://docs.oracle.com/javaee/7/api/javax/json/stream/package-summary.html is a good example of this pattern